gnucash/gnome-utils/dialog-account.c

```diff
@@ -1273,6 +1302,7 @@ commodity_changed_cb (GNCGeneralSelect *gsl, gpointer data)
 
     selection = gtk_tree_view_get_selection (GTK_TREE_VIEW (aw->transfer_tree));
     gtk_tree_selection_unselect_all (selection);
+    gnc_account_opening_balance_button_update (aw, currency);
```

Comment:
This could be a problem: What if the user changes the commodity on an account with the opening_balance flag set and there's already another account of that same commodity with the flag set?

Reply:
In that case the commodity change could not be processed and the user should be informed about this issue.

Reply:
And that happens where?

Reply:
in commodity_changed_cb(), see updated patch